src/sdk/modules/validators/smartSessionValidator/smartSessionValidator.test.ts

```diff
@@ -0,0 +1,324 @@
+    })
+    expect(realSig).toBeDefined()
+  })
+
+  test("should create Counter increment session (USE mode) on installed smart session validator", async () => {
+    const isInstalledBefore = await nexusClient.isModuleInstalled({
+      module: {
+        type: "validator",
+        address: addresses.SmartSession
+      }
+    })
+
+    expect(isInstalledBefore).toBe(true)
+
+    // make EOA owner of SA session key as well
+    const sessionKeyEOA = eoaAccount.address
+
+    // Todo: Add a negative test case for time range policy
+    const sessionRequestedInfo: CreateSessionDataParams = {
+      sessionPublicKey: sessionKeyEOA,
+      sessionValidatorAddress: TEST_CONTRACTS.SimpleSessionValidator.address,
+      sessionKeyData: toHex(toBytes(sessionKeyEOA)),
+      sessionValidAfter: 0,
+      sessionValidUntil: 0,
+      actionPoliciesInfo: [
+        {
+          contractAddress: TEST_CONTRACTS.Counter.address, // counter address
+          functionSelector: "0x273ea3e3" as Hex, // function selector for increment count
+          validUntil: 0, // 1717001666
+          validAfter: 0,
+          rules: [], // no other rules and conditions applied
+          valueLimit: BigInt(0)
+        }
+      ]
+    }
+
+    const smartSessionNexusClient = nexusClient.extend(
+      smartSessionValidatorActions()
+    )
+
+    const enableSessionsResponse = await smartSessionNexusClient.enableSessions(
+      {
+        account: nexusClient.account,
+        sessionRequestedInfo: [sessionRequestedInfo]
+      }
+    )
+
+    expect(enableSessionsResponse.userOpHash).toBeDefined()
+    expect(enableSessionsResponse.permissionIds).toBeDefined()
+
+    const permissionIds = enableSessionsResponse.permissionIds
+    expect(permissionIds.length).toBe(1)
+    const permissionId = permissionIds[0]
+    cachedPermissionId = permissionId
+
+    const receipt = await nexusClient.waitForUserOperationReceipt({
+      hash: enableSessionsResponse.userOpHash
+    })
+
+    expect(receipt.success).toBe(true)
+
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: permissionId
+    })
+    expect(isEnabled).toBe(true)
+  }, 60000)
+
+  test("should make use of already enabled session (USE mode) to increment a counter using a session key", async () => {
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: cachedPermissionId
+    })
+    expect(isEnabled).toBe(true)
+
+    const smartSessionValidator = await toSmartSessionValidatorModule({
+      client: nexusClient.account.client as PublicClient,
+      initData: encodePacked(["address"], [eoaAccount.address]),
+      deInitData: "0x",
+      nexusAccountAddress: nexusClient.account.address,
+      activePermissionId: cachedPermissionId
```

Comment:
This shouldn't be necessary either right? The permissionId is passed in  when `useEnabledSession` is called. No need to keep the activePermissionId here.

Reply:
recently I saw some dapp's implementation of passkeyModule and they set all is needed when initing module and then keep using it.

Reply:
I made it optional then it seems like 
smartSessionValidator.activePermissionId = permissionId has no effect inside useSession